app/models/manageiq/providers/vmware/infra_manager/vim_connect_mixin.rb

```diff
@@ -53,6 +53,11 @@ def raw_connect(options)
       validate_connection do
         vim = MiqVim.new(ip, user, pass)
         raise MiqException::Error, _("Adding ESX/ESXi Hosts is not supported") unless vim.isVirtualCenter
+
+        # If the time on the vCenter/Host is very far off from MIQ system time then
+        # any comparison of last_refresh_on and VMware TaskInfo.completeTime can be
+        # unreliable.
+        raise MiqException::Error, _("Time is too far out of sync") if (Time.parse(vim.currentTime) - Time.now.utc).abs > 10.minutes
```

Comment:
Should we be clearer that it is the vcenter time that is off?  Or perhaps that the "vcenter time is too far out of sync with the ManageIQ system time" (or something like that)

Reply:
Also is vim.currentTime in utc or not? (I'm not sure it even matters because with time math the time zones don't really matter)

Reply:
yes currentTime is UTC (you can't actually set a timezone on vCenter afaik, everything is utc)

```
vim.currentTime
"2020-09-09T21:43:55.832282Z"
```